integrations/shopify/src/index.ts

```diff
@@ -63,6 +63,26 @@ export class Shopify implements TriggerIntegration {
       throw `Can't create Shopify integration (${options.id}) as apiKey was undefined`;
     }
 
+    if (Object.keys(options).includes("apiSecretKey") && !options.apiSecretKey) {
+      throw `Can't create Shopify integration (${options.id}) as apiSecretKey was undefined`;
+    }
+
+    if (Object.keys(options).includes("adminAccessToken") && !options.adminAccessToken) {
+      throw `Can't create Shopify integration (${options.id}) as adminAccessToken was undefined`;
+    }
+
+    if (Object.keys(options).includes("hostName") && !options.hostName) {
+      throw `Can't create Shopify integration (${options.id}) as hostName was undefined`;
+    }
+
+    // Regular expression to ensure the hostname ends with `.myshopify.com`
+    const shopifyDomainPattern = /^[a-zA-Z0-9-]+\.myshopify\.com$/;
+
+    // Verify that the user has entered the .myshopify.com domain and not a custom primary domain
+    if (!shopifyDomainPattern.test(options.hostName)) {
+      throw `Can't create Shopify integration (${options.id}) because hostName should be a ".myshopify.com" domain, not a custom primary domain.`;
```

Comment:
[According to the shopify docs
](https://shopify.dev/docs/api/admin-rest#endpoints)
> All Admin REST API endpoints follow this pattern:
> https://{store_name}.myshopify.com/admin/api/2024-04/{resource}.json
If OAuth will ever be used then i think hostName can be a custom domain, but from my experience with shopify rest api, if you dont use the stores .myshopify.com domain (initial one when created, so even if you change your .myshopify.com domain) [you will get a 301 redirect with no info about your current request.](https://community.shopify.com/c/graphql-basics-and/custom-domain-to-call-api/m-p/2387000)

If backwards compatibility is an issue we can remove this check completly and have it mentioned somewhere in the integration docs. Trailing slashes are a no go. Their specs are mentioned [here](https://github.com/Shopify/shopify-app-js/blob/main/packages/apps/shopify-api/docs/reference/shopifyApi.md)

But from experience things can get pretty confusing. If we want to allow custom domains we would have to pass a `customShopDomains`. 

tbh spending too much time on this is not even worth it as shopify is planning and will release new features only on the GraphQL API (for example if you want to access the new 2k variant limit instead of 100 you will have to use graphql api)

Even without the checks people should still get the error message logged when response.ok fails so it should be more than enough to debug it

TL/DR: i can remove it completly and maybe mention somewhere in the docs (not necessary tho)

Reply:
I'm done with the changes but i also tested if i use another .myshopify.com domain (you can change it once for every store i think). The webhook appears as created but it's not registering correctly.

![image](https://github.com/triggerdotdev/trigger.dev/assets/55840806/a3ad9780-4d99-475b-8aba-6666271197ab)
![image](https://github.com/triggerdotdev/trigger.dev/assets/55840806/452fa18a-1f9d-46a9-ba04-01a99bf3de26)

It's because it always returns a 301 redirect. i can also update it to throw an error warning the user to use the primary domain. My guess would be where we check for res.ok 

```
 if (!response.ok) {
          await handleWebhookError("WEBHOOK_CREATE", request, response);
}
```

I'm not sure if this is the actual best place to put that check.

Reply:
![image](https://github.com/triggerdotdev/trigger.dev/assets/55840806/e16e0e7a-9782-4570-a71f-832dbd82fd2c)

I think this should do?